app/controllers/search_controller.rb

```diff
@@ -2,8 +2,15 @@ class SearchController < ApplicationController
   skip_before_filter :authenticate_user!, :only => [ :result ]
   def result
     # First search in morphology mode, if fails — retry in wildcard mode
-    @projects = Project.search params[:q]
-    @projects = Project.search params[:q], star: true if @projects.empty?
+    
+    if params[:episode] == "all"
```

Comment:
This is already done for all actions in all controllers because of the [set_episode](https://github.com/SUSE/hackweek/blob/master/app/controllers/application_controller.rb#L7) [filter](http://guides.rubyonrails.org/action_controller_overview.html#filters) in `ApplicationController`.

Reply:
That is true, but everytime the user hits the search button the params hash resets to contain only the search query. Since `:episode` key does not exist, according to the code in `ApplicationController`, `@episode` always returns the active hackweek. So I cannot search selectively unless I somehow get `:episode` in the hash.

Reply:
Ah true, this is it's own route. Then have a look at how the project routes are handled with the [named parameter scope](http://guides.rubyonrails.org/routing.html#prefixing-the-named-route-helpers).

Reply:
Do you want me to add the search route to the `:episode` scope?

Reply:
Yes

Reply:
the `:episode` in the project route is optional. I tried adding it there but the search is always ignoring `:episode` and theres no episode id in the params. If I don't make it optional, it doesn't work for pages like the homepage and sign in because they don't have `:episode` in their params. I get this
`ActionController::UrlGenerationError - No route matches {:action=>"result", :controller=>"search"} missing required keys: [:episode]:`

Reply:
I fixed it. It should work now

Reply:
I fixed it. Should work now.
@hennevogel Please review.